### PR TITLE
GH-1444: Observablility Documentation, Polish

### PR DESCRIPTION
**Pull request by garyrussell**
Refactor for latest snapshots; add documentation generation. Use NOOP registry and supplier for context.

Also fix class tangles.

Resolves https://github.com/spring-projects/spring-amqp/issues/1507

Disable auto doc generation and polish manually.

### CODE REVIEW

File: spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/RabbitListenerTests.java
Comment (artembilan) [inline spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/RabbitListenerTests.java:103]: I don't think we need this since this PR is going to be merged already after respective change in the SF.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java:2439]: With the latest Observation SNAPSHOT we don't need an explicit cast - they just remove the method with `Context`. Just `Supplier` is possible from now on.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java:1545]: DITTO

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitListenerObservation.java
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitListenerObservation.java:89]: FYI: I have fixed this tangle just using literals instead of enums: https://github.com/spring-projects/spring-integration/pull/3896/files#diff-1966bee1c00ff5bd693214849eb988472f9f86f986023f598cc8d344a7740afbR39

**Review by artembilan — CHANGES_REQUESTED** [review]
Also you have many failed tests in GH actions report.

File: spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/RabbitListenerTests.java
Comment (garyrussell) [reply spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/RabbitListenerTests.java:103]: We need it (temporarily) so that the PR build will succeed.

**Review by artembilan — APPROVED** [review]
LGTM.
Will merge when greed GH actions build.

**Comment by garyrussell** [comment]
Green finally; will address the logback and web client issues after merge.

**Pull request merged**
